dashboard_viewer/uploader/file_handler/checks.py

```diff
@@ -311,3 +318,100 @@ def check_for_duplicated_files(uploaded_file, data_source_id):
         pass
 
     #### Added For Checksum ##########################
+
+def upload_data_to_tmp_table(data_source_id, file_metadata, pending_upload):
+    # Upload New Data to a "temporary" table
+    pending_upload.uploaded_file.seek(0)
+
+    reader = pandas.read_csv(
+        pending_upload.uploaded_file,
+        header=0,
+        dtype=file_metadata["types"],
+        skip_blank_lines=False,
+        index_col=False,
+        names=file_metadata["columns"],
+        chunksize=500,
+    )
+
+    all_mat_views = MaterializedQuery.objects.exclude(matviewname__contains="tmp")
```

Comment:
What happens if two users are uploading data at the same time?

Reply:
Hm... Good Point. Although I did not face any issue in dev, I think is better to place the whole process in a lock, to prevent unexpected behavior.